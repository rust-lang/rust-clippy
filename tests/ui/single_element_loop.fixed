//@run-rustfix
// Tests from for_loop.rs that don't have suggestions

#[warn(clippy::single_element_loop)]
fn main() {
    let item1 = 2;
    {
        let item = &item1;
        dbg!(item);
    }

    {
        let item = &item1;
        dbg!(item);
    }

    {
        let item = &(0..5);
        dbg!(item);
    }

    {
        let item = &mut (0..5);
        dbg!(item);
    }

    {
        let item = 0..5;
        dbg!(item);
    }

    {
        let item = 0..5;
        dbg!(item);
    }

    // should not lint (issue #10018)
    for e in [42] {
        if e > 0 {
            continue;
        }
    }

    // should not lint (issue #10018)
    for e in [42] {
        if e > 0 {
            break;
        }
    }

    // should lint (issue #10018)
    {
        let _ = 42;
        let _f = |n: u32| {
            for i in 0..n {
                if i > 10 {
                    dbg!(i);
                    break;
                }
            }
        };
    }
}
