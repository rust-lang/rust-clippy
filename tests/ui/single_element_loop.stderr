error: for loop over a single element
  --> $DIR/single_element_loop.rs:8:5
   |
LL | /     for item in &[item1] {
LL | |         dbg!(item);
LL | |     }
   | |_____^
   |
   = note: `-D clippy::single-element-loop` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::single_element_loop)]`
help: try
   |
LL ~     {
LL +         let item = &item1;
LL +         dbg!(item);
LL +     }
   |

error: for loop over a single element
  --> $DIR/single_element_loop.rs:12:5
   |
LL | /     for item in [item1].iter() {
LL | |         dbg!(item);
LL | |     }
   | |_____^
   |
help: try
   |
LL ~     {
LL +         let item = &item1;
LL +         dbg!(item);
LL +     }
   |

error: this loops only once with `item` being `0..5`
  --> $DIR/single_element_loop.rs:16:17
   |
LL |     for item in &[0..5] {
   |                 ^^^^^^^ help: did you mean to iterate over the range instead?: `0..5`

error: this loops only once with `item` being `0..5`
  --> $DIR/single_element_loop.rs:20:17
   |
LL |     for item in [0..5].iter_mut() {
   |                 ^^^^^^^^^^^^^^^^^ help: did you mean to iterate over the range instead?: `0..5`

error: this loops only once with `item` being `0..5`
  --> $DIR/single_element_loop.rs:24:17
   |
LL |     for item in [0..5] {
   |                 ^^^^^^ help: did you mean to iterate over the range instead?: `0..5`

error: this loops only once with `item` being `0..5`
  --> $DIR/single_element_loop.rs:28:17
   |
LL |     for item in [0..5].into_iter() {
   |                 ^^^^^^^^^^^^^^^^^^ help: did you mean to iterate over the range instead?: `0..5`

error: for loop over a single element
  --> $DIR/single_element_loop.rs:47:5
   |
LL | /     for _ in [42] {
LL | |         let _f = |n: u32| {
LL | |             for i in 0..n {
LL | |                 if i > 10 {
...  |
LL | |         };
LL | |     }
   | |_____^
   |
help: try
   |
LL ~     {
LL +         let _ = 42;
LL +         let _f = |n: u32| {
LL +             for i in 0..n {
LL +                 if i > 10 {
LL +                     dbg!(i);
LL +                     break;
LL +                 }
LL +             }
LL +         };
LL +     }
   |

error: aborting due to 7 previous errors

